src/vuln_analysis/utils/source_rpm_downloader.py

```diff
@@ -0,0 +1,731 @@
+        Args:
+            filename: RPM filename
+            
+        Returns:
+            dict: Parsed package info or None if invalid
+        """
+        if not filename.endswith('.src.rpm'):
+            return None
+
+        #filesystem-3.16-5.el9.src.rpm
+        #fonts-rpm-macros-2.0.5-7.el9.1.src.rpm    
+        #libsemanage-3.6-5.el9_6.src.rpm
+        #python-setuptools-53.0.0-13.el9_6.1.src.rpm
+
+
+        # Remove .src.rpm extension
+        base_name = filename[:-8]
+                   
+
+        # Split by dashes to get parts
+        parts = base_name.split('-')
+        if len(parts) < 3:
+            return None
+        #fonts,rpm,macros,2.0.5,7.el9.1
+        if len(parts) == 3:
+            package_name = parts[0]            
+        elif len(parts) > 3:            
+        # Build package name from all parts except last 2
+            package_name_parts = []
+            for i in range(len(parts) - 2):
+               package_name_parts.append(parts[i])
+            package_name = '-'.join(package_name_parts)        
+            # Last 2 elements: version and platform
+        version_part = parts[-2]
+        full_platform = parts[-1]
+        
+        #cases 
+        #5.el9
+        #7.el9.1
+        #5.el9_6
+        #13.el9_6.1
+        platform = None
+        if 'el' in full_platform:
+            # Extract base platform by removing _ and . and finding el followed by digits
+            import re
+            match = re.search(r'el\d+', full_platform)
+            if match:
+                platform = match.group()
+        else:
+            platform = full_platform
+
+        return {
+            'name': package_name,
+            'version': version_part,
+            'release': full_platform,
+            'platform': platform,
+            'filename': filename
+        }
+
+    def _build_cache_index(self):
+        """
+        Build cache index from existing RPM files in cache directory.
+        
+        Returns:
+            dict: Cache index with key format: (package_name, version, platform)
+        """
+        if self.rpm_cache_dir is None:
+            return {}
+            
+        cache_index = {}
+        
+        for rpm_file in self.rpm_cache_dir.glob("*.src.rpm"):
+            parsed = self._parse_rpm_filename(rpm_file.name)
+            if parsed:
+                # Key: (package_name, version, platform)
+                key = (parsed['name'], parsed['version'], parsed['platform'])
+                cache_index[key] = {
+                    'filepath': rpm_file,
+                    'release': parsed['release'],
+                    'filename': parsed['filename']
+                }
+                #logger.debug(f"Cached RPM: {parsed['name']}-{parsed['version']}-{parsed['platform']}")
+        
+        logger.info(f"Cache index built: {len(cache_index)} RPMs found")
+        return cache_index
+
+    def _get_cache_path(self, package):
```

Comment:
@RedTanny Please use type hints/annotation for an increased readability and maintainability.